File: src/test/java/randomidprovider/BSNGeneratorTest.java
package randomidprovider;

import org.junit.Test;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

/**
 *
 */
public class BSNGeneratorTest {
    @Test
    public void testGetBsn() throws Exception {
        BSNGenerator bsnGenerator = new BSNGenerator(0);

        for (int i =0; i<100;i++) {
            String bsn = bsnGenerator.getBsn();
            System.out.println(bsn);
            assertThat(bsnGenerator.isElfProof(bsn), is(true));
        }
    }

    @Test
    public void testIsElfProof() {
        BSNGenerator bsnGenerator = new BSNGenerator();
        assertThat(bsnGenerator.isElfProof("73.61.60.221"), is(true));
    }

    @Test
    public void testSwapIsNotElfProof() {
        BSNGenerator bsnGenerator = new BSNGenerator();
        assertThat(bsnGenerator.isElfProof("73.16.60.221"), is(false));
    }

    @Test
    public void wrongNumberIsNotElfProof() {
        BSNGenerator bsnGenerator = new BSNGenerator();
        assertThat(bsnGenerator.isElfProof("73.62.60.221"), is(false));
    }
}

File: Generator/src/test/scala/RandomIdProvider/Entities/BSNGeneratorTest.scala
package RandomIdProvider.Entities

import junit.framework.{TestCase, TestSuite, Test}
import junit.framework.Assert._
import org.junit
import junit.runner.RunWith

import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.FlatSpec
import org.scalatest.matchers.ShouldMatchers


import util.Random
import scalax.logging.Logging
import RandomIdProvider.generator.BSNGenerator


/**
 * Unit test for simple App.
 */
@RunWith(classOf[JUnitRunner])
class BSNGeneratorTest extends FlatSpec with ShouldMatchers {


  "The BSNGenerator" should "generate 100 correct numbers" in {
    val bsnGenerator = new BSNGenerator(new Random(0))
    for (i <- 0 to 100) {
      val bsn = bsnGenerator.getBsn
      bsnGenerator.isElfProof(bsn) should equal (true)
    }

  }

  it should "validate a correct bsn number" in {
    val bsnGenerator = new BSNGenerator()
    bsnGenerator.isElfProof("73.61.60.221") should equal (true)
  }

  it should "not validate a swap" in {
    val bsnGenerator = new BSNGenerator()
    bsnGenerator.isElfProof("73.16.60.221") should equal (false)
  }

  it should "not validate a wrong number" in {
    val bsnGenerator = new BSNGenerator()
    bsnGenerator.isElfProof("73.62.60.221") equals (false)
  }

}
